src/runtime/CollectionWrappers/IterableWrapper.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class IterableWrapper<T> : IEnumerable<T>
```

Comment:
To be a lossless conversion, when passed back to Python an instance of this class should pass the `pyObject` field. I think this behavior must be added along with decoders, that create it.

Reply:
@lostmsu Right now I only defined decoders.  I would think this is the responsibility of the code handling object conversion back to python (i.e. encoders) - not the collection wrappers themselves.

In this branch these decoders are used by default, but in a future PR which replaces implicit copy list conversion with them, we can make sure to handle this.